packages/features/profile/repositories/ProfileRepository.ts

```diff
@@ -435,10 +464,46 @@ export class ProfileRepository {
     return profile;
   }
 
-  static async findByUpId(upId: string) {
+  static async findByUid(uid: string) {
+    const profile = await prisma.profile.findFirst({
+      where: {
+        uid,
+      },
+      include: {
+        user: {
+          select: userSelect,
+        },
+        movedFromUser: {
+          select: {
+            id: true,
+          },
+        },
+        organization: {
+          select: organizationWithSettingsAndMembersSelect,
+        },
+      },
+    });
+
+    if (!profile) {
+      return null;
+    }
+
+    return normalizeProfile(profile);
+  }
+
+  static async findByUpId(upId: string, userId: number) {
     const lookupTarget = ProfileRepository.getLookupTarget(upId);
-    log.debug("findByUpId", safeStringify({ upId, lookupTarget }));
+    log.debug("findByUpId", safeStringify({ upId, lookupTarget, userId }));
+
     if (lookupTarget.type === LookupTarget.User) {
+      // For user profiles, only allow access if userId matches
+      if (lookupTarget.id !== userId) {
+        log.warn(
+          "Unauthorized access attempt to user profile",
+          safeStringify({ upId, userId, targetUserId: lookupTarget.id })
+        );
+        return null;
+      }
```

Comment:
Why are we limiting this to just logged in user? even org owner/admins should be able to access user's profile